### PR TITLE
feat(route): add GitHub Pulse

### PR DESCRIPTION
**Pull request by jameschensmith**
## Involved Issue / 该 PR 相关 Issue

Not Applicable.

## Example for the Proposed Route(s) / 路由地址示例

```routes
/github/pulse/ziglang/zig
/github/pulse/DIYgod/RSSHub/daily
/github/pulse/DIYgod/RSSHub/halfweekly
/github/pulse/DIYgod/RSSHub/weekly
/github/pulse/DIYgod/RSSHub/monthly
/github/pulse/DIYgod/RSSHub/should-default-to-weekly
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [v2 Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [v2 路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [x] Documentation / 文档说明
  - [x] EN / 英文文档
  - [ ] CN / 中文文档
- [ ] Full text / 全文获取
  - [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制 
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施? 
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

For [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date), I use the current date / time the pulse page was called since the page records details from the current date. If there are other suggestions on what to do here, I welcome all feedback. Thanks! Excited to be contributing this ☺️

### CODE REVIEW

File: website/docs/routes/programming.md [hunk 1]
```diff
@@ -316,6 +316,10 @@ For instance, the `/github/topics/framework/l=php&o=desc&s=stars` route will gen
 
 <Route author="hashman TonyRL" example="/github/pull/DIYgod/RSSHub" path="/github/pull/:user/:repo/:state?/:labels?" paramsDesc={['User name', 'Repo name', 'the state of pull requests. Can be either `open`, `closed`, or `all`. Default: `open`.', 'a list of comma separated label names']} radar="1" rssbud="1"/>
 
+### Repo Pulse {#github-repo-pulse}
+
+<Route author="jameschensmith" example="/github/pulse/DIYgod/RSSHub" path="/pulse/:user/:repo/:period?" paramsDesc={['User name', 'Repo name', 'Time frame, selected from a repository\'s Pulse/Insights page. Possible values are: `daily`, `halfweekly`, `weekly`, or `monthly`. Default: `weekly`.']} radar="1" rssbud="1"/>
```
Comment (TonyRL) [inline website/docs/routes/programming.md:321]:
```suggestion
<Route author="jameschensmith" example="/github/pulse/DIYgod/RSSHub" path="/github/pulse/:user/:repo/:period?" paramsDesc={['User name', 'Repo name', 'Time frame, selected from a repository\'s Pulse/Insights page. Possible values are: `daily`, `halfweekly`, `weekly`, or `monthly`. Default: `weekly`.']} radar="1" rssbud="1"/>
```

File: lib/v2/github/pulse.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                    heading: $section.text(),
+                    paragraph: $paragraph.length > 0 ? $paragraph.text() : undefined,
+                    items: $list
+                        .children()
+                        .map((_, item) => {
+                            const $item = $(item);
+                            const $link = $item.find('a');
+                            return {
+                                link: { text: $link.text(), url: $link.attr('href') },
+                                details: $item.find('p').text(),
```
Comment (TonyRL) [inline lib/v2/github/pulse.js:57]: Consider showing the `title` attr of `relative-time` element here.

File: lib/v2/github/pulse.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        link,
+        item: [
+            {
+                title: periodHeading,
+                description: art(path.join(__dirname, 'templates/pulse-description.art'), {
+                    overviewItems,
+                    commitActivity,
+                    githubActivity,
+                }),
+                pubDate: new Date().toUTCString(),
```
Comment (TonyRL) [inline lib/v2/github/pulse.js:77]: Do not add a fake pubDate here. You can use the beginning of `periodHeading` as pubDate.

Please add a guid. The default `title` has a high chance of collision. You can try something like `md5(user, repo, period, periodHeading)`

Comment (jameschensmith) [reply lib/v2/github/pulse.js:77]: You want GUID in the title? Just making sure I understand where you'd like to have the GUID. Thanks!

File: lib/v2/github/pulse.js (hunk 2 above)
Comment (jameschensmith) [reply lib/v2/github/pulse.js:57]: Great idea!

Comment (jameschensmith) [reply lib/v2/github/pulse.js:57]: Ah, ran into an issue, `relative-time` does not have `title` attribute when scraping. It does have `datetime` attribute, which we could parse and use that instead.

File: lib/v2/github/pulse.js (hunk 3 above)
Comment (TonyRL) [reply lib/v2/github/pulse.js:77]: https://www.rssboard.org/rss-specification#ltguidgtSubelementOfLtitemgt
https://docs.rsshub.app/joinus/advanced/advanced-feed

**Pull request merged**